From 6ff6724a7e949fdf336921f63d5365bdcf7a265e Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 16 Sep 2024 11:42:12 +0200
Subject: [PATCH 1217/1695] plat-stm32mp1: conf: get HUK location from the DT
 if not given in command line

When CFG_STM32MP15_HUK is enabled and CFG_STM32MP15_HUK_OTP_BASE nor
CFG_STM32MP15_HUK_BSEC_KEY_0/_1/_2/_3 are defined in commande line,
enable by default CFG_STM32_HUK_FROM_DT to read the HUK location from
the device tree.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git bbb9b83b922158ffb884542b53fa373a036709a4 plat-stm32mp1: conf: get HUK location from the DT if not given in command line]
Change-Id: Ia31b2d0c2fa1c90a98ceff2dd73fa18865aa615f
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404582
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 27 ++++++---------------------
 1 file changed, 6 insertions(+), 21 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 697ea69b4..cd186dccf 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -44,20 +44,6 @@ flavorlist-512M = $(flavorlist-cryp-512M) \
 flavorlist-1G = $(flavorlist-cryp-1G) \
 		$(flavorlist-no_cryp-1G)
 
-flavorlist-MP15-HUK-DT = $(flavor_dts_file-157A_DK1) \
-			 $(flavor_dts_file-157A_ED1) \
-			 $(flavor_dts_file-157A_EV1) \
-			 $(flavor_dts_file-157C_DK2) \
-			 $(flavor_dts_file-157C_ED1) \
-			 $(flavor_dts_file-157C_EV1) \
-			 $(flavor_dts_file-157D_DK1) \
-			 $(flavor_dts_file-157D_ED1) \
-			 $(flavor_dts_file-157D_EV1) \
-			 $(flavor_dts_file-157F_DK2) \
-			 $(flavor_dts_file-157F_ED1) \
-			 $(flavor_dts_file-157F_EV1)
-
-
 flavorlist-MP15 = $(flavor_dts_file-157A_DHCOR_AVENGER96) \
 		  $(flavor_dts_file-157A_DK1) \
 		  $(flavor_dts_file-157A_ED1) \
@@ -162,12 +148,6 @@ ifeq ($(CFG_STM32MP_PROFILE),system_services)
 include $(platform-dir)/conf.disable-secure-services.mk
 endif
 
-# Some platform do not support some resources
-ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-MP15-HUK-DT)),)
-CFG_STM32MP15_HUK ?= y
-CFG_STM32_HUK_FROM_DT ?= y
-endif
-
 ifneq ($(filter $(CFG_EMBED_DTB_SOURCE_FILE),$(flavorlist-no_cryp)),)
 $(call force,CFG_STM32_CRYP,n)
 $(call force,CFG_STM32_PKA,n)
@@ -537,7 +517,12 @@ CFG_STM32_EARLY_CONSOLE_UART ?= 4
 # content derived with the device UID fuses content. See derivation scheme
 # in stm32mp15_huk.c implementation.
 CFG_STM32MP15_HUK ?= n
-CFG_STM32_HUK_FROM_DT ?= n
+ifeq (,$(CFG_STM32MP15_HUK_BSEC_KEY_0)$(CFG_STM32MP15_HUK_OTP_BASE))
+# Default get the HUK location from the DT if not specified in the configuration
+CFG_STM32_HUK_FROM_DT ?= y
+else
+$(call force,CFG_STM32_HUK_FROM_DT,n)
+endif
 
 ifeq ($(CFG_STM32MP15_HUK),y)
 ifneq ($(CFG_STM32_HUK_FROM_DT),y)
-- 
2.47.2

